Pod/Classes/SEGAdobeIntegration.m

```diff
@@ -372,17 +372,59 @@ - (void)trackHeartbeatEvents:(SEGTrackPayload *)payload
             return;
         }
         self.ADBMediaHeartbeat = [self.ADBMediaHeartbeatFactory createWithDelegate:nil andConfig:self.config];
-        self.mediaObject = [self.ADBMediaObjectFactory createWithProperties:payload.properties];
-
-        NSMutableDictionary *standardVideoMetadata = [self mapStandardVideoMetadata:payload.properties];
-        [self.mediaObject setValue:standardVideoMetadata forKey:ADBMediaObjectKeyStandardVideoMetadata];
-
+        self.mediaObject = [self createMediaObject:payload.properties];
         //TODO: check to see if we need to handle custom metadata (second argument) like with
         // contextDataVariables
         [self.ADBMediaHeartbeat trackSessionStart:self.mediaObject data:payload.properties];
         SEGLog(@"[ADBMediaHeartbeat trackSessionStart:%@ data:@%]", self.mediaObject, payload.properties);
         return;
     }
+
+    if ([payload.event isEqualToString:@"Video Playback Paused"]) {
+        [self.ADBMediaHeartbeat trackPause];
+        SEGLog(@"[ADBMediaHeartbeat trackPause]");
+        return;
+    }
+
+    if ([payload.event isEqualToString:@"Video Playback Buffer Started"]) {
```

Comment:
The only difference in some of these is the `ADBMediaHeartbeatEvent....` part. Think this should be abstracted so we can re-use some of the shared code.